doc/contribute/external.rst

```diff
@@ -0,0 +1,168 @@
+.. _external-contributions:
+
+Contributing source code from external projects
+***********************************************
+
+In some cases it is desirable to leverage existing, external source code in
+order to avoid re-implementing basic functionality or features that are readily
+available in other open source projects.
+
+This section describes the circumstances under which external source code can be
+imported into Zephyr, and the process that governs the inclusion.
+
+There are three main factors that will be considered during the inclusion
+process in order to determine whether it will be accepted. These will be
+described in the following sections.
+
+Software License
+================
+
+.. note::
+
+   External source code licensed under the Apache-2.0 license is not subject to
+   this section.
+
+Integrating code into the Zephyr Project from other projects that use a license
+other than the Apache 2.0 license needs to be fully understood in
+context and approved by the `Zephyr governing board`_, as described in the
+`Zephyr project charter`_. The board will automatically reject licenses that
+have not been approved by the `Open Source Initiative (OSI)`_. See the
+:ref:`external-src-process` section for more details.
```

Comment:
Note: do we need to care if this concerns content that does not end up in the firmware built?